lib/internal/abort_controller.js

```diff
@@ -357,6 +362,24 @@ function transferableAbortController() {
   return AbortController[kMakeTransferable]();
 }
 
+/**
+ * @param {AbortSignal} signal
+ * @param {any} resource
+ * @returns {Promise<void>}
+ */
+function aborted(signal, resource = null) {
+  try {
+    abortSignalValidator(signal, 'signal');
+  } catch (err) {
+    return PromiseReject(err);
+  }
+  if (signal.aborted)
+    return PromiseResolve();
+  const abortPromise = createDeferredPromise();
+  signal.addEventListener('abort', abortPromise.resolve, { [kWeakHandler]: resource, once: true });
```

Comment:
Semantically - I'm wondering what the signature should be. If resource defaults to `null` I'm not sure what the semantics even are - the listener may be cleaned up immediately and the promise would forever pend.

Reply:
I think could replace with if null then make it depend on the signal?

Reply:
That's a good question, caution suggests we start with making it required.

Reply:
Ah ok making resource compulsory

Reply:
Done!